src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -926,25 +969,29 @@ protected virtual string GetOutputFileName(Compilation compilation, Cancellation
         }
         private Func<string, FileMode, FileAccess, FileShare, Stream> _fileOpen;
 
-        private Stream OpenFile(string filePath,
-                                TextWriter consoleOutput,
-                                FileMode mode = FileMode.Open,
-                                FileAccess access = FileAccess.ReadWrite,
-                                FileShare share = FileShare.None)
+        private void ReportExceptionAsDiagnostic(Exception e, string filePath, TextWriter consoleOutput)
```

Comment:
A better question is why is this method even an instance method on Compilation? It's there probably because it was just used by Compilation. Now it's used in two places. Perhaps the method should be on CommonMessageProvider. Or it can just be static. Lambdas are too big a hammer for this.